eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/ReadRepoInfoFromSourceManifest.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.UnifiedBuild.Tasks
+{
+    // Takes a path to a source-manifest.json file and
+    // reads the information for a specific repo from it.
+    public class ReadRepoInfoFromSourceManifest : Task
+    {
+        [Required]
+        public string SourceManifest { get; set; } = "";
+
+        [Required]
+        public string RepositoryName { get; set; } = "";
+
+        /// <summary>
+        /// Returns metadata about the repository from source-manifest.json:
+        /// - PackageVersion
+        /// - BarId
+        /// - RemoteUri
+        /// - CommitSha
+        /// </summary>
+        [Output]
+        public ITaskItem? RepoInfo { get; set; }
+
+        public override bool Execute()
+        {
+            if (!File.Exists(SourceManifest))
+            {
+                Log.LogError($"Source manifest file not found: {SourceManifest}");
+                return false;
+            }
+
+            JsonArray? repositories = JsonNode.Parse(File.OpenRead(SourceManifest))?["repositories"]?.AsArray();
+
+            JsonNode? repo = repositories
+                ?.Where(p => p?["path"]?.ToString() == RepositoryName)
+                .FirstOrDefault()
+                ?.AsObject();
+
+            if (repo == null)
+            {
+                Log.LogError($"Repository {RepositoryName} not found in source manifest.");
+                return false;
+            }
+
+            RepoInfo = new TaskItem(
+                        RepositoryName,
+                        new Dictionary<string, string?>
+                        {
+                            ["PackageVersion"] = repo?["packageVersion"]?.ToString(),
+                            ["BarId"] = repo?["barId"]?.ToString(),
+                            ["RemoteUri"] = repo?["remoteUri"]?.ToString(),
+                            ["CommitSha"] = repo?["commitSha"]?.ToString()
```

Comment:
Do we need to hardcode the list here? This is in reaction to https://github.com/dotnet/dotnet/commit/9a0f13b9e9f8d4ac10ad119d90dfc387cb120e9f which removed the hardcoded metadata when transforming xml -> msbuild.

Reply:
yeah good idea